### PR TITLE
frontend: npm build fix

### LINKED ISSUE #83: Investigate why the frontend can't build with the client dependency
```
cd frontend
npm run build
```

the process will error and exit

### PR DESCRIPTION
**Pull request by bwdmonkey**
:tickets: **Ticket(s)**: Closes #83

---

## :construction_worker: Changes

- **Undo npm eject**
- Kept SCSS support
- Kept react-router support
- Kept linter support

## :flashlight: Testing Instructions

```
make deps
make web
(cd frontend/; npm run build)
```

## Versions
```bash
$ node --version
v11.6.0
$ npm --version
6.5.0-next.0
```

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -3,106 +3,37 @@
...
   },
   "scripts": {
-    "start": "node scripts/start.js",
-    "build": "node scripts/build.js",
+    "start": "react-scripts start",
+    "build": "react-scripts build",
     "preinstall": "npm install ../client/",
-    "test": "node scripts/test.js --env=jsdom",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject",
```
Comment (bobheadxi) [inline frontend/package.json]:
```suggestion
```

dont tempt me

File: frontend/package.json [hunk 2]
```diff
@@ -3,106 +3,37 @@
...
-    "postcss-flexbugs-fixes": "3.2.0",
-    "postcss-loader": "2.0.8",
-    "promise": "8.0.1",
-    "prop-types": "15.6.2",
-    "raf": "3.4.0",
-    "react": "^16.5.2",
-    "react-dev-utils": "^5.0.2",
-    "react-dom": "^16.5.2",
+    "prop-types": "^15.6.2",
+    "react": "^16.7.0",
```
Comment (bobheadxi) [inline frontend/package.json]: can we lock all this down to specific versions?

```suggestion
    "react": "16.7.0",
```

File: frontend/src/components/Reset.js [hunk 3]
```diff
@@ -6,11 +6,11 @@ class Reset extends Component {
     this.state = {
       email: '',
     };
-    this.updatetextfields = this.updatetextfields.bind(this);
+    this.updateTextFields = this.updateTextFields.bind(this);
     this.attemptSendReset = this.attemptSendReset.bind(this);
   }
 
-  updatetextfields(e) {
+  updateTextFields(e) {
```
Comment (bobheadxi) [inline frontend/src/components/Reset.js]: whee thanks

**Review by bobheadxi — CHANGES_REQUESTED** [review]
fantastic work! a few small changes

can you also enable this? https://github.com/ubclaunchpad/pinpoint/blob/master/.travis.yml#L68

File: frontend/package.json (hunk 1 above)
Comment (leesw98) [reply frontend/package.json]: 🤣

File: frontend/package.json (hunk 2 above)
Comment (leesw98) [reply frontend/package.json]: TODO: tomorrow do this

**Review by bobheadxi — APPROVED** [review]
Terrific, thanks!

**Pull request merged**